test/sdf_test.cpp

```diff
@@ -42,7 +42,7 @@ TEST(SDF, SphereShell) {
       },
       {vec3(-1.1), vec3(1.1)}, 0.01, 0, 0.0001);
 
-  EXPECT_NEAR(sphere.Genus(), 11500, 1000);
+  EXPECT_NEAR(sphere.Genus(), 14235, 1000);
```

Comment:
Does this still look good, or look better? I recall working to remove little bits of infinitely thin geometry on this one.

Reply:
I don't see infinitely thin stuff, but I see something that looks like self-intersection:

![image](https://github.com/user-attachments/assets/ae074d8c-61bd-47f4-b185-40fff3ba34e4)

This is probably related to the other thing I mentioned, as there are many  non-2-manifold edges here.

Reply:
Ah, right - I think my slick pre-edge collapse thing in the SDF can create even manifold edges. We should probably look at that again and see if there's a better way. I believe without that it's all guaranteed 2-manifold.